hadoop-hdds/common/src/test/java/org/apache/hadoop/hdds/fs/TestDedicatedDiskSpaceUsage.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hdds.fs;
+
+import org.apache.hadoop.fs.FileUtil;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+
+import static org.apache.hadoop.hdds.fs.TestDU.createFile;
+import static org.apache.hadoop.test.GenericTestUtils.getTestDir;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Tests for {@link DedicatedDiskSpaceUsage}.
+ */
+public class TestDedicatedDiskSpaceUsage {
+
+  private static final File DIR =
+      getTestDir(TestDedicatedDiskSpaceUsage.class.getSimpleName());
+
+  private static final int FILE_SIZE = 1024;
+
+  @Before
+  public void setUp() {
+    FileUtil.fullyDelete(DIR);
+    assertTrue(DIR.mkdirs());
+  }
+
+  @After
+  public void tearDown() throws IOException {
+    FileUtil.fullyDelete(DIR);
+  }
+
+  @Test
+  public void testGetUsed() throws IOException {
+    File file = new File(DIR, "data");
+    createFile(file, FILE_SIZE);
+    SpaceUsageSource subject = new DedicatedDiskSpaceUsage(DIR);
+
+    assertTrue(subject.getUsedSpace() >= FILE_SIZE - 20);
```

Comment:
Can you please add a comment explaining -20 for margin of error.

Reply:
It comes from [`TestDFCachingGetSpaceUsed`](https://github.com/apache/hadoop/blob/2aa065d98f36527d7769c9c58a923a706036391d/hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/fs/TestDFCachingGetSpaceUsed.java#L61) from Hadoop. :)